arch/risc-v/src/mpfs/mpfs_opensbi_utils.S

```diff
@@ -0,0 +1,275 @@
+/****************************************************************************
+ * arch/risc-v/src/mpfs/mpfs_opensbi_utils.S
+ *
+ * Adapted for NuttX from the logic on OpenSBI fw_base.S
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright (c) 2019 Western Digital Corporation or its affiliates and other
+ * contributors.
```

Comment:
we can add Western Digital to LICENSE file, and apply Apache license here

Reply:
ok!

Reply:
I don't think we can apply Apache license here.  We have to keep the original license and at the same time add Western Digital to the LICENSE file.